syncer/dml.go

```diff
@@ -74,24 +83,31 @@ func (c *GenColCache) clearTable(schema, table string) {
 	key := dbutil.TableName(schema, table)
 	delete(c.hasGenColumn, key)
 	delete(c.columns, key)
-	delete(c.indexes, key)
 	delete(c.isGenColumn, key)
 }
 
 func (c *GenColCache) reset() {
 	c.hasGenColumn = make(map[string]bool)
 	c.columns = make(map[string][]*column)
-	c.indexes = make(map[string]map[string][]*column)
 	c.isGenColumn = make(map[string][]bool)
 }
 
-func genInsertSQLs(schema string, table string, dataSeq [][]interface{}, columns []*column, indexColumns map[string][]*column) ([]string, [][]string, [][]interface{}, error) {
+func genInsertSQLs(param *genDMLParam) ([]string, [][]string, [][]interface{}, error) {
+	var (
+		schema               = param.schema
+		table                = param.table
+		dataSeq              = param.data
+		originalDataSeq      = param.originalData
+		columns              = param.columns
+		originalColumns      = param.originalColumns
+		originalIndexColumns = param.originalIndexColumns
+	)
 	sqls := make([]string, 0, len(dataSeq))
 	keys := make([][]string, 0, len(dataSeq))
 	values := make([][]interface{}, 0, len(dataSeq))
 	columnList := genColumnList(columns)
 	columnPlaceholders := genColumnPlaceholders(len(columns))
-	for _, data := range dataSeq {
+	for dataIdx, data := range dataSeq {
 		if len(data) != len(columns) {
```

Comment:
also check `originalColumns ` and `originalDataSeq `?

Reply:
not needed, data/columns and origianlData/originalColumns have the same length distance ?

Reply:
add column and data length check in generated column prune